index.js

```diff
@@ -1,8 +1,43 @@
-import Redis from 'redis';
+import { createClient } from 'redis';
+
+export const generateConnectUrl = (config = {}) => {
+    let auth = '';
+    let connection = `${config.host}:${config.port ?? 6379}`;
+    let db = '';
+    if (config.username || config.password) {
+        auth = `${config.user ?? ''}${config.password !== undefined ? `:${config.password}` : ''}@`;
+    }
+    if (config.db !== undefined) {
+        db = `/${config.db}`;
+    }
+    return `redis://${auth}${connection}${db}`;
+}
```

Comment:
I propose using the [web compatible `URL` module](https://nodejs.org/api/url.html). It handles URL encoding and is a bit terser.
I noticed that you used both `config.user` and `config.username`. The latter is the correct one.

```suggestion
export const generateConnectUrl = (config) => {
    const connectionUrl = new URL('https://example.com')
    connectionUrl.host = config.host
    connectionUrl.port = `${config.port ?? 6379}`
    connectionUrl.username = config.username ?? ''
    connectionUrl.password = config.password ?? ''
    connectionUrl.pathname = config.db ?? ''
    connectionUrl.protocol = 'redis:'
    return connectionUrl.toString()
}
```

Reply:
@MatthiasKunnen updated to use URL with a few minor changes

Reply:
Regarding the changes you made:

- Setting the protocol to 'redis:' in the URL constructor will actually not work in browsers, see here: https://codesandbox.io/s/browser-url-protocol-9hmgwq?file=/src/index.ts
- Adding the ifs makes it less readable and repeats the property, I'm not sure why you prefer it over `?? ''`
- The leading `/` for the `pathname` is not necessary
- I do not see why `config` should have a default parameter since all calls to it provide it?